### DISCUSSION TITLE
Are cookies used by Supabase Auth exempt from GDPR, or do I need to use a banner notifying the users that we're storing cookies?

### DISCUSSION THREAD

**Discussion by lumenwrites** [supabase/supabase#7738] [Questions]
We're using Supabase Auth on our website. Do you guys know whether we have to add a GDPR banner notifying the user that we use cookies?

**Comment by b3nten** [comment]
"Under the EU's GDPR, cookies that are not strictly necessary for the basic function of your website must only be activated after your end-users have given their explicit consent to the specific purpose of their operation and collection of personal data."

**Reply by lumenwrites** [reply]
Thank you for your reply, but unfortunately it doesn't answer my question.

Do the cookies used by Supabase Auth require a banner? Login functionality is necessary for the function of my website, and we don't store any other kind of cookies. But it is unclear whether the cookies that supabase stores are exempt from GDPR or not.

**Reply by b3nten** [reply]
Supabase Auth doesn't use cookies, it stores the session in local storage. If you're setting cookies for Supabase Auth, for example with Supabase Auth Helpers, it's up to you to determine if they are necessary for the functionality of the site. My guess is that they would be, but it depends what they're used for. 

Sorry I can't give a yes/no.

**Comment by Nemmcy** [comment]
I had the same question and, after reading all the relevant parts of the GDPR law, I concluded that the data contained in "{currentSession: {,…}, expiresAt: ...}" and stored by Supabase Auth in local storage is **necessary** for the provision of the service.

Regarding the location of storage, there is no real difference between storing data in cookies or in local storage, the question is WHAT data is stored and HOW it is used.

Here is a very user-friendly [summary by Iubenda](https://www.iubenda.com/en/help/7515-which-data-can-be-safely-stored-in-cookies-without-having-to-request-consent) (they have a very strong expertise in data privacy, it's their business).

No need for consent if:
- Technical cookies that are solely necessary for the provision of the service. (eg. preference cookies, session cookies, load balancing, etc.)
- Statistical cookies managed directly by you (not third-parties), provided that the data is not used for [profiling](https://gdpr-info.eu/art-22-gdpr/).

So the type of data and the way it is stored by Supabase Auth does not require a consent from the user, but you have to be careful about how you use such data! I believe, but I am not sure, that simply [assessing or classifying users](https://iapp.org/news/a/wp29-releases-guidelines-on-profiling-under-the-gdpr/#:~:text=The%20working%20party%20emphasizes%20that,with%20or%20without%20predictive%20purpose.), i.e. creating different PostgreSQL tables, based on some of the data contained in currentSession, such as email or phone, is profiling and would require a consent.

Note that I was only referring to Supabase Auth, not the company Supabase Inc. Does Supabase Inc, via their hosting service, collect/store/use such data in a way that our app would only be compliant with the GDPR law **if and only if** we (Supabase's customers) ask the consent from our users (the users of Supabase's customers) before storing the session data in local storage? I don't know.

I'm not a compliance expert, it's just my personal understanding of the laws.
